### ISSUE TITLE
GIS: Add dialog box to set maximum one-way hours when loading GIS data into a project

### ISSUE THREAD

**Issue by lbross** [USFS-PNW/Fia-Biosum-Manager#374]
The units will be the same as the travel_time.one_way_hours field. For example, 2 hours and 30 minutes will be 2.5. Add simple validation to ensure input is a double value. Set default value to 2.

**Comment by lbross** [comment]
I found that the control I was planning to use for this does not exist in C# :-( I was remembering it from my Visual Basic days. VB has a control that is similar to a MessageBox except that you can add a single text box for input. In C# you have to create a custom form to receive any input. If I have to put in the effort to set that up, I might as well create a form that covers the series of Message Boxes that the GIS Load process currently uses. This is what I came up with. We can change the text if you want, but let's try to keep it simple.
![image](https://github.com/user-attachments/assets/20425c3e-a612-4b4c-9ed0-2b4bfe8209eb)


I will hide checkboxes 1 and 2 if there is no existing GIS data in the project. I will hide checkbox 4 if the field isn't in the plot_gis table that can be optionally used to update the plot yarding distance. I will copy the logic over from the current button which hopefully won't take too long.
Let me know what you think.

**Comment by jsfried** [comment]
That looks peachy to me! Missing only a help button (for which i'll need to write some text concerning these very consequential decisions the user must make). A nice to have would be the display of the path to the GIS data about to be loaded and it's date stamp and size to confirm to the user that they are loading from a DB that they want to be loading from (e.g., if they make their own GIS travel times database, they would know what date and size that was and could confirm they are loading from that one vs a BioSum default)

**Comment by lbross** [comment]
I will make those changes. Let me know when you have the help text.

**Comment by jsfried** [comment]
@lbross Here is the help text:
Wood processing facilities and travel times to those facilities from each of the plots in your project can be loaded from a master_gis_travel_times.db SQLite database, provided that one exists in your user’s appdata BioSum folder (e.g., C:\Users\jsfried\AppData\Roaming\FIABiosum\ master_gis_travel_times.db). A version of this database can be obtained at http://biosum.info/
If your project already contains a gis_travel_times.db database and you want to proceed with a new load of GIS data, check the box to allow your existing data to be overwritten. If you’d like to save your existing gis_travel_times.db database for backup or archival purposes, check the box to create a backup. 
When loading from the master_gis_travel_times, what is the maximum one-way haul time to include in the hunt for potential destinations for harvested wood feedstocks? The default threshold is 2 hours, which can be modified if desired. A larger threshold will generate a local gis_travel_times.db database with the potential to deliver to more distant locations. This may be necessary in regions where wood processing infrastructure is sparse, else there may be no wood utilized; where such infrastructure is dense, setting a large threshold may generate a processing site list with hundreds of facilities to scroll through in Optimizer when selecting (via check box) which to include in analyses. Setting a low threshold may result in nonmerchantable wood (dirty chips) getting modeled as burned at the landing if that choice (of low threshold) results in the local gis_travel_times database containing no facilities to receive such material. 
Check the “update yarding distance”  box only if confident that the master_travel_times database contains a better representation of the distance to road than what is stored in the MoveDistance_ft field in the FIADB_FVS_VARIANT table in the biosum_ref database in the AppData\Roaming\FIABiosum folder – a value sourced as the distance class mid-point in feet associated with the PLOT.RDDISTCD value collected in the last field visit before creation of the FIADB_FVS_VARIANT table.

**Comment by lbross** [comment]
@jsfried I formatted the text to make it consistent with the rest of the help. Please review and give final approval. You should be able to download it from GitHub [here](https://github.com/USFS-PNW/Fia_Biosum_Setup/blob/master/Help/TREATMENT_OPTIMIZER_Help.docx). The GIS help is on the last page.

**Comment by jsfried** [comment]
@lbross Thanks for the formatting and constructive edits. Downloading was easy-- not sure how to edit on Github. I guess the only thing I question is this:
"There is also the option of populating the MoveDist_ft_REPLACEMENT field with distances using a third party gis process." Where is the MoveDist_ft_REPLACEMENT field? I don't see it in the master_gis_travel_times table on my computer or in the FIADB_FVS_VARIANT table in biosum_ref

It seems fine to say that users may wish to rely on a GIS process to determine proximity of plots to a road layer that they bring to the party, but should keep in mind that the available plot location data is fuzzed to be up to 0.71 miles from the actual locations, which is why by defauly, we rely on the crew-called distance to road represented in PLOT.RDCLASS

**Comment by lbross** [comment]
The background on this is that Sebastian wanted to be able to update the plot yarding distance from a plot_gis table that he had created. It was relatively easy to continue to support this, but we didn't think lots of analysts would want to do it, and we didn't want to confuse them. The MoveDist_ft_REPLACEMENT is NOT in the plot_gis table by default. The option to update the plot yarding distance from plot_gis doesn't appear on the UI if this field is missing. So most analysts will never know it is an option.
If we have another Sebastian, they can add the MoveDist_ft_REPLACEMENT field to plot_gis and either copy the values from MoveDist_ft_NEAR or come up with their own values another way. Perhaps this is too esoteric to be included in the Help and should just be in the User Guide?

**Comment by lbross** [comment]
If I delete the fourth paragraph about the plot_gis table, are we good to go? Can you check the GIS section of the User Guide and see if this is in there? If not, I can take a swing at writing it up. I'd hate for the option to get "lost" if someone needs it in the future.

**Comment by jsfried** [comment]
Good to go. Will check user guide. Adding to my to do list.
________________________________

**Comment by lbross** [comment]
The help button and text in BioSum are ready to go. The User Guides for loading GIS Data needs to be updated, but not until after the v5.11.2 release because the process is changing. There is a paragraph in the user guide about updating from plot_gis that also needs to be updated.

**Comment by Cat-Ducat-USDA** [comment]
Max hours box is present and in correct units and the correct combination of checkboxes is coming up when looking at an existing vs. new project (w/ or w/o previous GIS data uploaded). However, the help button is broken. When I clicked it, BioSum froze briefly and then threw an error message saying "An error occurred while trying to display the Help! Please contact the BioSum support."

**Comment by lbross** [comment]
This may be something I noticed in my testing. The first time you click the Help button, it works. But the second time, it fails. I'm not sure if it's the same Help button twice in a row or any two help buttons in the application. This hails back to a workaround I put in a few years ago when a Microsoft update broke our Help system that doesn't appear to work if the Help system is accessed more than once. I'll dig around a bit and see if there is a newer solution.

**Comment by lbross** [comment]
I replaced the workaround with the original built-in Microsoft function that was broken by a security patch in December 2022 and it appears to work again! See issue #293. This change will be in the final v5.11.2 .msi.

**Comment by Cat-Ducat-USDA** [comment]
Sounds good! So it should be working now? I will test it. It didn't work for me at all, even the first time I clicked it yesterday.

**Comment by jsfried** [comment]
Probably becasue you had already popped a help screen elsewhere in BioSum before trying to summon that one. I was successful on another help screen for the first go round but not the 2nd.

**Comment by Cat-Ducat-USDA** [comment]
Hmm, could be. I don't recall using the Help button at all prior to that, but who knows! I just tried it on the new project and it worked the first time, but not the second time, as well.

**Comment by lbross** [comment]
This fix requires a new .msi so it won't work correctly until that is distributed.

**Comment by lbross** [comment]
Fix to help handled under issue #293. All else is good.
